### PR TITLE
docs: rename auth user to auth identity

### PR DESCRIPTION
**Pull request by shahednasser**
Change all occurrences of auth user to auth identity

### CODE REVIEW

File: www/apps/book/app/advanced-development/api-routes/protected-routes/page.mdx
Comment (adrien2p) [inline www/apps/book/app/advanced-development/api-routes/protected-routes/page.mdx]: todo: The line bellow should reference store/customers/me instead of store/me no?

Comment (shahednasser) [reply www/apps/book/app/advanced-development/api-routes/protected-routes/page.mdx]: Not occurding to this: https://github.com/medusajs/medusa/blob/ebc4215f16acc288a0e57e4f6f4c820d2891cb02/packages/medusa/src/loaders/helpers/routing/index.ts#L345

**Comment by shahednasser** [comment]
@olivermrbl can we get this one in?

File: www/apps/book/app/advanced-development/api-routes/protected-routes/page.mdx
Comment (shahednasser) [reply www/apps/book/app/advanced-development/api-routes/protected-routes/page.mdx]: @adrien2p also wanna point out that this has been the behavior for a while (see our [v1 docs](https://docs.medusajs.com/development/api-routes/create#protect-store-api-routes)), but I do agree with you it should be `store/customers/me`

Comment (adrien2p) [reply www/apps/book/app/advanced-development/api-routes/protected-routes/page.mdx]: Sorry I was a bit ahead I think https://github.com/medusajs/medusa/pull/7396/files#diff-43af01e7f3abbcfc6c4c34d314fbac19fe975994c517304ed00678c5f3830d6fR338 😂

Comment (shahednasser) [reply www/apps/book/app/advanced-development/api-routes/protected-routes/page.mdx]: Ah ok I didn't see that PR 😄 then the authenticate middleware now applies on all `/store` routes (just without requiring it) right? It doesn't seem we apply anything for `/store/customers/me` anymore

File: www/apps/resources/app/commerce-modules/auth/examples/page.mdx
Comment (sradevski) [inline www/apps/resources/app/commerce-modules/auth/examples/page.mdx]:
```suggestion
      auth_identities: await authModuleService.list(),
```

Made the same find-and-replace mistake 😄 There are few more occurences like that.

File: www/apps/book/app/advanced-development/api-routes/protected-routes/page.mdx
Comment (adrien2p) [reply www/apps/book/app/advanced-development/api-routes/protected-routes/page.mdx]: yes it seems correct it has probably changed again, you can have a look with stev he might be on it at the moment

File: www/apps/resources/app/commerce-modules/auth/persisting-auth-user/page.mdx
Comment (sradevski) [inline www/apps/resources/app/commerce-modules/auth/persisting-auth-user/page.mdx:56]: `auth` is now `auth_context`

File: www/apps/book/app/advanced-development/api-routes/protected-routes/page.mdx
Comment (sradevski) [inline www/apps/book/app/advanced-development/api-routes/protected-routes/page.mdx]: We moved away from using `scope` (as that usually means something different in the authz world) to using `Actor type`

File: www/apps/resources/app/commerce-modules/auth/persisting-auth-user/page.mdx
Comment (sradevski) [inline www/apps/resources/app/commerce-modules/auth/persisting-auth-user/page.mdx:24]: We don't need to do it now, but I feel we can provide better explanation here.

The auth Identity is basically an identifier of an actor - where an actor can be user, customer, driver, a backend service, etc. It doesn't really carry any details of the actor itself (eg. first name, last name), but it is only used to identify them.

File: www/apps/resources/app/commerce-modules/auth/examples/page.mdx
Comment (shahednasser) [reply www/apps/resources/app/commerce-modules/auth/examples/page.mdx]: Haha nice catch 😄

File: www/apps/resources/app/commerce-modules/auth/persisting-auth-user/page.mdx
Comment (shahednasser) [reply www/apps/resources/app/commerce-modules/auth/persisting-auth-user/page.mdx:24]: I'm actually removing this page in another PR as it felt the details in it weren't necessary (for their original purpose). However, it would make sense to have a new document around adding new actors, etc... (once we fully support that through custom links), so would hold off on this for now

Comment (sradevski) [reply www/apps/resources/app/commerce-modules/auth/persisting-auth-user/page.mdx:24]: @shahednasser you can create new actors, the old way, see: https://medusacorp.slack.com/archives/C025KMS13SA/p1716371728051229.

Now, the approach could change in the upcoming months, so I leave it up to you if it's time to document it or not 😄

Comment (shahednasser) [reply www/apps/resources/app/commerce-modules/auth/persisting-auth-user/page.mdx:24]: I could be misunderstanding, but isn't the way of linking the actor types (for example, driver) still in progress or done in a not-so-ideal way? If that's the case then yea I'd rather wait on documenting it. But if it's something different please let me know.

Comment (sradevski) [reply www/apps/resources/app/commerce-modules/auth/persisting-auth-user/page.mdx:24]: @shahednasser it is done (eg. Victor has used it successfully in Medusa Eats), and whether it is ideal or not is something we haven't decided yet 😄

Comment (shahednasser) [reply www/apps/resources/app/commerce-modules/auth/persisting-auth-user/page.mdx:24]: The only reason I say this is because I remember when we discussed with @olivermrbl how to get custom links working we decided we'd wait on sharing it until we've improved the developer experience of it, not sure if I missed another part of the discussion though 😄

Comment (sradevski) [reply www/apps/resources/app/commerce-modules/auth/persisting-auth-user/page.mdx:24]: @shahednasser yeah that's right, but for auth we are not using links at all now. Whether we'll want to switch back to links is something we'll need to decide later on, but for now you simply set eg. `user_id` in `app_metadata` on the auth identity and that's it (that acts like a link basically)

Comment (shahednasser) [reply www/apps/resources/app/commerce-modules/auth/persisting-auth-user/page.mdx:24]: Ahh ok I wasn't aware of that. Can you share a document or link to a slack chat around the approach now? I believe I missed it

Comment (sradevski) [reply www/apps/resources/app/commerce-modules/auth/persisting-auth-user/page.mdx:24]: @shahednasser I was gonna write an internal doc on how auth works, I'll have it with you in ~30 min

Comment (shahednasser) [reply www/apps/resources/app/commerce-modules/auth/persisting-auth-user/page.mdx:24]: @sradevski cool no rush, I won't include it in this PR anyway 👍🏻

**Comment by shahednasser** [comment]
@olivermrbl can we get this PR in?

**Review by olivermrbl — APPROVED** [review]

**Review by adrien2p — APPROVED** [review]

**Pull request merged**
